frontend/src/plugins/impl/DataTablePlugin.tsx

```diff
@@ -90,20 +95,26 @@ export const DataTablePlugin = createPlugin<S>("marimo-table")
             nulls: z.number().nullish(),
             true: z.number().nullish(),
             false: z.number().nullish(),
-          }),
+          })
         ),
-      }),
+      })
     ),
+    sort_values: rpc
+      .input(z.object({ by: z.string().nullable(), descending: z.boolean() }))
+      .output(z.union([z.string(), z.array(z.object({}).passthrough())])),
   })
   .renderer((props) => {
     if (typeof props.data.data === "string") {
+      const [sorting, setSorting] = useState<SortingState>([]);
```

Comment:
does this fail if you put this in `LoadingDataTableComponent`? `useState` shouldnt be in a conditional, and should be in a React component

Reply:
I put this outside of `Loading` so that I could pass it as a prop, since `memo` requires props to change in order for the component to re-render

Reply:
can't remember why I had it there